### PR TITLE
Add response methods for read_xml and read_html

### PR DESCRIPTION
**Pull request by jimhester**
Also convert read_html to a generic.

Perhaps the `need_package("xml")` is unnecessary, as how do you get a response object without having httr installed, but I put it in there to be safe.

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -20,3 +20,13 @@ quote_str <- function(x, quote = "\"") {
 
   paste0(quote, x, quote)
 }
+
+is_installed <- function(pkg) {
+  system.file(package = pkg) != ""
```
Comment (hadley) [inline R/utils.R]: Can we use `requireNamespace(quietly = TRUE)` instead?

Comment (jimhester) [inline R/utils.R]: Sure (this is originally from https://github.com/hadley/httr/blob/master/R/utils.r#L35)

Comment (hadley) [inline R/utils.R]: You should never trust that guy's code!

**Comment by hadley** [comment]
LGTM

**Pull request merged**